plugins/src/main/java/com/google/firebase/gradle/plugins/report/TestReportGenerator.kt

```diff
@@ -373,5 +360,30 @@ class TestReportGenerator(private val apiToken: String) {
 
   companion object {
     private const val URL_PREFIX = "https://api.github.com/repos/firebase/firebase-android-sdk/"
+    private const val GITHUB_API_VERSION = "2022-11-28"
+    // Pulls the URL corresponding to the rel="next" link header, if present.
+    // Ignores other link header values ("prev", etc) and ignores parameters
+    // eg `<http://www.foo.bar/>; baz="qux"; rel="next";` -> `http://www.foo.bar/`
+    private val NEXT_LINK_REGEX =
+      Regex(
+        "<" + // eg `<http://www.foo.bar/>`
+          "(" + // URL group
+          /**/ "[^>]*" + // Only ignoring `>`, other illegal characters assumed not present
+          ")" +
+          ">" +
+          "\\s*" +
+          ";" + // Link separator
+          "(" + // Ignore other parameters, eg `foo="bar";`
+          /**/ "\\s*" +
+          /**/ "\\w+" + // Key
+          /**/ "=" +
+          /**/ "\"\\w*\"" + // Quoted value
+          /**/ "\\s*" +
+          /**/ ";" +
+          ")" +
+          "\\s*" +
+          "rel=\"next\""
```

Comment:
what about

```
        "<([^>]*)>" + // Capture the URL (group 1)
          "\\s*;" + // Separator
          "(?:\\s*\\w+=\"[^\"]*\"\\s*;)*" + // Non-capturing group for other parameters (e.g., baz="qux";)
          "\\s*rel=\"next\"" // Match rel="next"
```

It still documents the regex but keeps groups together for easier reading